core/preview.ts

```diff
@@ -75,7 +75,15 @@ export const previewCandidates = computed($candidates, candidates => {
 })
 
 onMount($candidates, () => {
-  return $links.listen(() => {})
+  let unsubscribeRouter = router.subscribe(page => {
+    if (page.route === 'add') {
+      setPreviewUrl(page.params.url!)
```

Comment:
We already have `setPreviewUrl()` call on `router` changes https://github.com/hplush/slowreader/blob/main/core/preview.ts#L319

Maybe it will be better to fix that block to avoid code duplication?

Reply:
Can I ask you to do a deep investigation:

1. Add `console.log` with different variables to that `router.listen`
2. Reproduce an issue
3. Find a reason why exactly the previous code didn’t work
4. Explain it in fix PR

It is very dangerous to fix an issue without going deep into why it happened.

We need something like:
> We have a issue because `previewUrl` was set on page loading event, so I added extra variable to track page loading event